### PR TITLE
feat(orm): add typed QuerySet date projections

### LINKED ISSUE #5848: Feature: Add typed QuerySet date projections
## Prerequisites

- [x] I have searched existing issues to verify this feature hasn't been requested

## Summary

Add typed `dates` and `datetimes` QuerySet projections with deterministic truncation, time-zone, distinctness, and ordering semantics.

Split from #5812. Target branch: `develop/0.4.0`.

## Motivation / Problem

Applications currently reimplement backend-sensitive date truncation and distinct projection. The ORM needs one typed contract that defines week boundaries, time-zone conversion order, and unsupported backend behavior.

## Proposed Solution

- Make `dates` accept a generated typed date field and return `Vec<chrono::NaiveDate>`.
- Make `datetimes` accept a generated typed datetime field and return values in the selected `chrono_tz::Tz`; use UTC by default.
- Support year, month, ISO week, and day for dates.
- Additionally support hour, minute, and second for datetimes.
- Convert to the requested time zone before truncation.
- Define ISO weeks as Monday-based.
- Perform distinct projection and ascending or descending ordering in the database.
- Return a capability error when a backend cannot correctly perform a requested named-time-zone conversion.
- Never silently use UTC or the server-local time zone for an unsupported request.
- Add compile-fail field tests, cross-backend truncation tests, DST gap/fold and ISO week/year boundary tests, and user-facing documentation.

### Durable implementation and verification contracts

- Represent truncation, time-zone conversion, distinct projection, and ordering with typed `reinhardt-query` statements and expressions rather than unchecked SQL strings in QuerySet code.
- Provide caller-owned-executor variants wherever equivalent QuerySet execution paths support them. If no equivalent path exists, document that non-applicability instead of inventing an unrelated executor API.
- Add cross-backend tests for every date and datetime truncation kind in both ascending and descending order.
- Add time-zone and projection tests for UTC, a non-UTC named zone, DST gaps and folds, ISO week/year boundaries, null exclusion, and distinct values.
- Update the relevant crate README, `lib.rs` module documentation, and ORM user-facing documentation. Include typed-field and caller-owned-executor examples where applicable, state backend time-zone limitations explicitly, and document the intentionally stricter-than-Django capability error instead of silent UTC or server-local fallback.

## Alternatives Considered

1. Truncate in Rust after loading all source values. This moves an unbounded projection out of the database.
2. Use the server-local time zone. This makes results deployment-dependent.
3. Add a global ORM time zone in the same issue. This broadens the configuration scope unnecessarily.

## Is this a breaking change?

No - backward compatible

## Migration Path (if breaking)

Not applicable. These are additive projection APIs.

## Additional Context

Global ORM time-zone configuration is explicitly deferred. This issue excludes retrieval helpers, streaming iteration, explain, and row locking.

### PR DESCRIPTION
**Pull request by kent8192**
## Summary

This PR addresses:

- Adds typed `QuerySet::dates` and `QuerySet::datetimes` projections for generated date and UTC datetime fields.
- Performs truncation, null exclusion, distinctness, and ordering in PostgreSQL, MySQL, and SQLite SQL.
- Applies named time zones before PostgreSQL truncation and returns explicit capability errors for unsupported MySQL/SQLite named-zone requests.
- Adds Manager, caller-owned connection, and active transaction entry points plus backend, DST, ISO-week, runtime, and compile-fail coverage.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (code change that neither fixes a bug nor adds a feature)
- [x] Documentation update
- [ ] Performance improvement
- [x] Code quality improvements
- [ ] CI/CD changes
- [ ] Other (please describe):

## Breaking Change Assessment

- [x] **No** - This change is backward compatible
- [ ] **Yes** - This change breaks existing public API or behavior

## Motivation and Context

Date and datetime bucketing previously required raw SQL or client-side post-processing. The new structural expression and typed generated-field API keep backend differences inside the query layer while preventing non-temporal fields from compiling.

Fixes #5848

Related to: #5812

## How Was This Tested?

- `cargo test -p reinhardt-query --test temporal_projection -- --test-threads=1` (3 passed)
- `cargo test -p reinhardt-db --no-default-features --features orm,sqlite --test date_projection -- --test-threads=1` (5 passed)
- `cargo test -p reinhardt-db --no-default-features --features orm,sqlite,migrations --test date_projection_ui -- --test-threads=1` (pass and compile-fail fixtures passed)
- `cargo clippy -p reinhardt-query --all-features --lib --test temporal_projection -- -D warnings`
- `cargo clippy -p reinhardt-db --no-default-features --features orm,sqlite,migrations --lib --test date_projection --test date_projection_ui -- -D warnings`
- `cargo doc -p reinhardt-db --no-default-features --features orm,sqlite,migrations,associations --no-deps`
- `cargo fmt --all -- --check`
- `git diff --check`

PostgreSQL and MySQL are covered by exact checked SQL-generation assertions; SQLite additionally has an in-memory execution test. Live PostgreSQL/MySQL integration was not run locally.

## Checklist

- [x] I have followed the [Contributing Guidelines](../blob/main/CONTRIBUTING.md)
- [x] I have followed the [Commit Guidelines](../blob/main/instructions/COMMIT_GUIDELINE.md)
- [x] I have updated the documentation (if applicable)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have tested with all affected database backends (if applicable)
- [x] I have formatted the code with `cargo make fmt-fix`
- [ ] I have checked the code with `cargo make clippy-check`
- [ ] I use self-hosted runner for CI (Repository owner only)

## Related Issues

- Fixes #5848
- Related to #5812

### Type Label (select one)
- [ ] `bug` - Bug fix
- [x] `enhancement` - New feature or improvement
- [ ] `documentation` - Documentation update
- [ ] `performance` - Performance improvement
- [ ] `refactoring` - Code refactoring
- [ ] `code-quality` - Code quality improvements

### Additional Labels (apply alongside type label when applicable)
- [ ] `breaking-change` - Breaking change (**MUST** match "Yes" in Breaking Change Assessment above)

### Scope Label (select all that apply)
- [x] `database` - Database layer, schema, migrations
- [ ] `auth` - Authentication, authorization, sessions
- [x] `orm` - ORM layer, models, query builder
- [ ] `http` - HTTP layer, handlers, middleware
- [ ] `routing` - URL routing, path matching
- [ ] `api` - REST API, serializers, views
- [ ] `admin` - Admin interface, admin panels
- [ ] `forms` - Form handling, validation, rendering
- [ ] `graphql` - GraphQL schema, resolvers
- [ ] `websockets` - WebSocket connections, handlers
- [ ] `i18n` - Internationalization, localization
- [ ] `ci-cd` - CI/CD workflow changes

### Priority Label (for maintainers)
- [ ] `critical` - Blocks release or major functionality
- [x] `high` - Important fix or feature
- [ ] `medium` - Normal priority
- [ ] `low` - Minor fix or enhancement

---

**Additional Context:**

- Named-zone conversion is intentionally capability-gated instead of silently falling back to UTC or server-local time.

🤖 Generated with [Codex](https://openai.com/codex)

### CODE REVIEW

File: crates/reinhardt-db/src/orm/query.rs [hunk 1]
```diff
@@ -1382,6 +1479,183 @@ where
...
+	fn temporal_projection_statement(
+		&self,
+		field: &str,
+		kind: TemporalTruncKind,
+		order: DateProjectionOrder,
+		time_zone: Option<TemporalTimeZone>,
+		output: TemporalTruncOutput,
+	) -> reinhardt_core::exception::Result<SelectStatement> {
+		let source = Expr::col(self.root_column_reference(field)).into_simple_expr();
+		let projection = Func::temporal_trunc(source.clone(), kind, time_zone, output);
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-db/src/orm/query.rs]: Implemented in a1b251cb44: root column references are now qualified whenever manual joins are present, preventing ambiguous temporal projections.\n\n---\n*Generated by Codex.*

File: crates/reinhardt-db/tests/ui/date_projection/pass.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#![allow(dead_code, unexpected_cfgs)]
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-db/tests/ui/date_projection/pass.rs]: Implemented in a1b251cb44: each lint allowance now has an English rationale specific to the trybuild fixture.\n\n---\n*Generated by Codex.*

File: crates/reinhardt-db/tests/date_projection.rs [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+			"SELECT DISTINCT DATE_TRUNC('year', \"event_date\")::date AS \"value\" ",
+			"FROM \"projection_events\" WHERE \"event_date\" IS NOT NULL ORDER BY ",
+			"\"value\" DESC"
+		)
+	);
+}
+
+#[tokio::test]
+#[cfg(feature = "sqlite")]
+async fn sqlite_executes_distinct_null_excluding_date_and_datetime_projections() {
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-db/tests/date_projection.rs:305]: Implemented in a1b251cb44: the global database-registration test now uses the named date_projection_database serial group.\n\n---\n*Generated by Codex.*

File: crates/reinhardt-query/src/expr/func.rs [hunk 4]
```diff
@@ -61,6 +61,21 @@ impl Func {
 	pub fn coalesce(exprs: Vec<SimpleExpr>) -> SimpleExpr {
 		SimpleExpr::FunctionCall("COALESCE".into_iden(), exprs)
 	}
+
+	/// Create a typed temporal truncation expression.
+	pub fn temporal_trunc(
+		expr: SimpleExpr,
+		kind: TemporalTruncKind,
+		time_zone: Option<TemporalTimeZone>,
+		output: TemporalTruncOutput,
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-query/src/expr/func.rs:74]: Implemented in 98af838215: temporal truncation now returns a checked error for hour, minute, and second units requested as a date, preventing backend renderer panics; the invalid combination is covered by a focused test.\n\n---\n*Generated by Codex.*

File: crates/reinhardt-query/src/error.rs [hunk 5]
```diff
@@ -436,6 +444,16 @@ fn validate_simple_expr(expr: &SimpleExpr, backend: &'static str) -> Result<(),
 		| SimpleExpr::AsEnum(_, expression)
 		| SimpleExpr::ExprAlias(expression, _)
 		| SimpleExpr::Cast(expression, _) => validate_simple_expr(expression, backend),
+		SimpleExpr::TemporalTrunc {
+			expr, time_zone, ..
+		} => {
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-query/src/error.rs:452]: Implemented in 37f6e467b0: checked query building now rejects directly constructed temporal truncations that combine a time unit with Date output, before backend rendering.\n\n---\n*Generated by Codex.*

File: crates/reinhardt-db/src/orm/query.rs [hunk 6]
```diff
@@ -1382,6 +1479,186 @@ where
...
+		time_zone: Option<TemporalTimeZone>,
+		output: TemporalTruncOutput,
+	) -> reinhardt_core::exception::Result<SelectStatement> {
+		let source = Expr::col(self.root_column_reference(field)).into_simple_expr();
+		let projection =
+			Func::temporal_trunc(source.clone(), kind, time_zone, output).map_err(|error| {
+				DatabaseError::new(DatabaseErrorKind::Unsupported, error.to_string())
+			})?;
+		let mut stmt = Query::select();
+		self.apply_model_from(&mut stmt);
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-db/src/orm/query.rs:1716]: Implemented in 37f6e467b0: date and datetime projection APIs now reject querysets sourced from a raw derived subquery rather than silently querying the model base table.\n\n---\n*Generated by Codex.*

File: crates/reinhardt-db/src/orm/query.rs [hunk 7]
```diff
@@ -1382,6 +1479,193 @@ where
...
+		stmt.expr_as(projection.clone(), Alias::new("value"));
+		self.apply_relation_joins(&mut stmt);
+		self.apply_manual_joins(&mut stmt);
+		if let Some(condition) = self.build_where_condition()? {
+			stmt.cond_where(condition);
+		}
+		stmt.and_where(source.is_not_null());
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-db/src/orm/query.rs:1723]: Implemented in 1959d57d15. Temporal projections now apply the QuerySet group-by fields and HAVING aggregate conditions before producing distinct buckets. The regression test covers grouped date projection with COUNT(*) > 1.

— Codex

File: crates/reinhardt-query/src/backend/mysql.rs [hunk 8]
```diff
@@ -92,6 +92,84 @@ impl MySqlQueryBuilder {
...
 
+	fn write_datetime_in_zone(
+		&self,
+		writer: &mut SqlWriter,
+		expr: &SimpleExpr,
+		time_zone: Option<&TemporalTimeZone>,
+	) {
+		writer.push("CONVERT_TZ(");
+		self.write_simple_expr(writer, expr);
+		writer.push(", '+00:00', ");
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-query/src/backend/mysql.rs:103]: No code change is needed here. Model-generated DateTime<Utc> fields use FieldType::TimestampTz, which maps to DATETIME on MySQL in crates/reinhardt-db/src/migrations/fields.rs. That type is stored and bound as a UTC wall time, matching the +00:00 source zone used by the temporal projection. Changing every pool session time zone would alter unrelated connection behavior without correcting this DATETIME path.

— Codex

File: crates/reinhardt-db/src/orm/query.rs [hunk 9]
```diff
@@ -1382,6 +1479,235 @@ where
...
+		for group_field in &self.group_by_fields {
+			stmt.group_by_col(self.root_column_reference(group_field));
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-db/src/orm/query.rs]: Resolved in 3e26428962. Temporal projections now reject grouped querysets before generating SQL, preventing invalid or nondeterministic temporal bucket queries.

— Codex

File: crates/reinhardt-db/src/orm/query.rs [hunk 10]
```diff
@@ -1382,6 +1479,235 @@ where
...
+		let source = Expr::col(self.root_column_reference(field)).into_simple_expr();
+		let projection =
+			Func::temporal_trunc(source.clone(), kind, time_zone, output).map_err(|error| {
+				DatabaseError::new(DatabaseErrorKind::Unsupported, error.to_string())
+			})?;
+		let mut stmt = Query::select();
+		self.apply_model_from(&mut stmt);
+		stmt.expr_as(projection.clone(), Alias::new("value"));
+		self.apply_relation_joins(&mut stmt);
+		self.apply_manual_joins(&mut stmt);
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-db/src/orm/query.rs:1719]: Resolved in 3e26428962. Temporal projections now reject querysets with LATERAL joins instead of silently omitting the join from their SQL.

— Codex

File: crates/reinhardt-query/src/backend/postgres.rs [hunk 11]
```diff
@@ -88,6 +88,57 @@ impl PostgresQueryBuilder {
...
+			writer.push(" AT TIME ZONE ");
+			let zone = match time_zone {
+				Some(TemporalTimeZone::Named(zone)) => zone.clone(),
+				Some(TemporalTimeZone::Utc) | None => "UTC".to_string(),
+			};
+			self.write_temporal_zone(writer, zone, unquoted);
+		}
+		writer.push(")");
+		match output {
+			TemporalTruncOutput::Date => writer.push("::date"),
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-query/src/backend/postgres.rs:130]: Resolved in 3e26428962. PostgreSQL checked SELECT builds now run the same structural validation as the other backends, including direct invalid temporal truncation ASTs.

— Codex

File: crates/reinhardt-db/src/orm/query.rs [hunk 12]
```diff
@@ -1382,6 +1479,246 @@ where
...
+			)
+			.into());
+		}
+		let source = Expr::col(self.root_column_reference(field)).into_simple_expr();
+		let projection =
+			Func::temporal_trunc(source.clone(), kind, time_zone, output).map_err(|error| {
+				DatabaseError::new(DatabaseErrorKind::Unsupported, error.to_string())
+			})?;
+		let mut stmt = Query::select();
+		self.apply_model_from(&mut stmt);
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-db/src/orm/query.rs:1716]: Implemented in 6ea4ecb416: temporal projections now reject CTE-backed querysets before execution, with a regression test.

---
*Generated by Codex.*

File: crates/reinhardt-query/src/backend/postgres.rs [hunk 13]
```diff
@@ -88,11 +88,63 @@ impl PostgresQueryBuilder {
...
+			}
+		}
+	}
+
 	/// Build a SELECT statement through the checked query-building API.
 	pub fn build_select_checked(
 		&self,
 		stmt: &SelectStatement,
 	) -> Result<(String, Values), crate::QueryBuildError> {
+		crate::error::validate_select_for_backend(stmt, "PostgreSQL")?;
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-query/src/backend/postgres.rs:147]: Implemented in 6ea4ecb416: PostgreSQL checked SELECT construction no longer invokes the generic validator, so pgvector SELECT expressions continue to build on PostgreSQL.

---
*Generated by Codex.*

File: crates/reinhardt-db/tests/ui/date_projection/wrong_field.rs [hunk 14]
```diff
@@ -0,0 +1,29 @@
+#![allow(dead_code, unexpected_cfgs)]
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-db/tests/ui/date_projection/wrong_field.rs]: Implemented in a4f2cc627a5ea399c6db7d0907a76c0a187ab6ce. The compile-fail fixture now documents why each lint allowance is necessary for direct trybuild compilation.

---
*Generated by Codex.*

File: crates/reinhardt-db/src/orm/query.rs [hunk 15]
```diff
@@ -1382,6 +1479,253 @@ where
...
+			.into());
+		}
+		if !self.lateral_joins.is_empty() {
+			return Err(DatabaseError::new(
+				DatabaseErrorKind::Unsupported,
+				"date and datetime projections are not supported on querysets with lateral joins",
+			)
+			.into());
+		}
+		if !self.group_by_fields.is_empty() {
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-db/src/orm/query.rs]: Implemented in a4f2cc627a5ea399c6db7d0907a76c0a187ab6ce. Temporal projections now reject querysets with HAVING conditions even without GROUP BY, with a regression test for that path.

---
*Generated by Codex.*

File: crates/reinhardt-query/src/backend/postgres.rs (hunk 13 above)
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-query/src/backend/postgres.rs:147]: Implemented in 1c25c82f73: PostgreSQL validation now preserves pgvector values and distance operators while retaining the shared temporal validation.

---
*Generated by Codex.*

File: crates/reinhardt-db/tests/ui/date_projection/wrong_field.stderr [hunk 16]
```diff
@@ -0,0 +1,73 @@
+error[E0277]: the trait bound `std::string::String: DateProjectionField` is not satisfied
+  --> tests/ui/date_projection/wrong_field.rs:21:4
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-db/tests/ui/date_projection/wrong_field.stderr]: Implemented in 1c25c82f73: refreshed the trybuild diagnostic locations to match the current fixture.

---
*Generated by Codex.*

File: crates/reinhardt-query/src/backend/postgres.rs [hunk 17]
```diff
@@ -783,6 +835,12 @@ impl PostgresQueryBuilder {
 				writer.push_identifier(&type_name.to_string(), |s| self.escape_iden(s));
 				writer.push(")");
 			}
+			SimpleExpr::TemporalTrunc {
+				expr,
+				kind,
+				time_zone,
+				output,
+			} => self.write_temporal_trunc(writer, expr, *kind, time_zone.as_ref(), *output, false),
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-query/src/backend/postgres.rs:848]: Implemented in 1c25c82f73: checked PostgreSQL INSERT, UPDATE, and DELETE builders now run shared expression validation, including invalid temporal truncation rejection.

---
*Generated by Codex.*

File: crates/reinhardt-db/src/orm/query.rs [hunk 18]
```diff
@@ -1382,6 +1479,253 @@ where
...
+		let mut stmt = Query::select();
+		self.apply_model_from(&mut stmt);
+		stmt.expr_as(projection.clone(), Alias::new("value"));
+		self.apply_relation_joins(&mut stmt);
+		self.apply_manual_joins(&mut stmt);
+		if let Some(condition) = self.build_where_condition()? {
+			stmt.cond_where(condition);
+		}
+		stmt.and_where(source.is_not_null());
+		for having_cond in &self.having_conditions {
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-db/src/orm/query.rs]: Implemented in 7062602. The temporal projection path now rejects HAVING before statement construction and no longer contains the unreachable aggregate/HAVING loop; see `crates/reinhardt-db/src/orm/query.rs:1674`.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-query/tests/temporal_projection.rs [hunk 19]
```diff
@@ -0,0 +1,223 @@
...
+				kind,
+				TemporalTruncOutput::DateTime,
+				Some(TemporalTimeZone::Utc),
+				order,
+			);
+			let (postgres_sql, postgres_values) = PostgresQueryBuilder
+				.build_select_checked(&statement)
+				.unwrap();
+			assert_eq!(postgres_values.len(), 2);
+			assert_eq!(postgres_sql.matches("DATE_TRUNC").count(), 1);
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-query/tests/temporal_projection.rs]: Implemented in 7062602. `datetime_projection_sql_covers_every_kind_and_order_for_all_backends` now asserts complete backend SQL for both ASC and DESC, including the final ORDER BY clause; see `crates/reinhardt-query/tests/temporal_projection.rs:170`.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-db/src/orm/query.rs [hunk 20]
```diff
@@ -5839,6 +6183,161 @@ where
 		stmt.to_owned()
 	}
 
+	/// Return distinct truncated values from a generated date field.
+	///
+	/// Truncation, `DISTINCT`, null exclusion, and ordering are performed by the
+	/// database. ISO weeks begin on Monday. Grouped querysets and querysets with
+	/// lateral joins are not supported.
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-db/src/orm/query.rs]: Implemented in 7062602. All QuerySet and Manager temporal projection entry points now document subqueries, CTEs, lateral joins, and grouped/HAVING querysets as unsupported; see `crates/reinhardt-db/src/orm/query.rs:6353`.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-query/src/backend/postgres.rs [hunk 21]
```diff
@@ -783,6 +838,12 @@ impl PostgresQueryBuilder {
...
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-query/src/backend/postgres.rs:848]: Implemented in 7062602. PostgreSQL checked CREATE TABLE and ALTER TABLE builders now call the shared structural validators before pgvector validation, with regression coverage for invalid direct temporal defaults; see `crates/reinhardt-query/src/backend/postgres.rs:156` and `crates/reinhardt-query/src/backend/postgres.rs:4952`.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-db/src/orm/query.rs [hunk 22]
```diff
@@ -1382,6 +1479,253 @@ where
...
+			Func::temporal_trunc(source.clone(), kind, time_zone, output).map_err(|error| {
+				DatabaseError::new(DatabaseErrorKind::Unsupported, error.to_string())
+			})?;
+		let mut stmt = Query::select();
+		self.apply_model_from(&mut stmt);
+		stmt.expr_as(projection.clone(), Alias::new("value"));
+		self.apply_relation_joins(&mut stmt);
+		self.apply_manual_joins(&mut stmt);
+		if let Some(condition) = self.build_where_condition()? {
+			stmt.cond_where(condition);
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-db/src/orm/query.rs:1721]: Implemented in 7062602. Root filter compilation now uses `has_joined_tables`, covering manual joins, and the temporal projection regression asserts the root column is qualified; see `crates/reinhardt-db/src/orm/query.rs:5310` and `crates/reinhardt-db/src/orm/query.rs:11978`.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-query/src/backend/postgres.rs [hunk 23]
```diff
@@ -88,11 +88,63 @@ impl PostgresQueryBuilder {
...
 
+	fn write_temporal_zone(&self, writer: &mut SqlWriter, zone: String, unquoted: bool) {
+		if unquoted {
+			writer.push("'");
+			writer.push(&zone.replace('\'', "''"));
+			writer.push("'");
+		} else {
+			writer.push_value(crate::value::Value::String(Some(Box::new(zone))), |index| {
+				self.placeholder(index)
+			});
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-query/src/backend/postgres.rs:99]: Implemented in 7062602. CREATE VIEW now inlines the embedded SELECT values through the shared SQL-literal path, so PostgreSQL temporal zones remain bound for ordinary SELECTs but are literals in view definitions; see `crates/reinhardt-query/src/backend/postgres.rs:2127` and `crates/reinhardt-query/tests/temporal_projection.rs:267`.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-query/src/backend/sqlite.rs [hunk 24]
```diff
@@ -55,6 +55,81 @@ impl SqliteQueryBuilder {
 		Self
 	}
 
+	fn write_monday_date(&self, writer: &mut SqlWriter, expr: &SimpleExpr) {
+		writer.push("DATE(");
+		self.write_simple_expr(writer, expr);
+		writer.push(", '-' || ((CAST(strftime('%w', ");
+		self.write_simple_expr(writer, expr);
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-query/src/backend/sqlite.rs]: Implemented in 7062602. SQLite and MySQL week truncation now derive the ISO-week Monday from one source evaluation, and exact SQL tests cover both lowerings; see `crates/reinhardt-query/src/backend/sqlite.rs:58` and `crates/reinhardt-query/src/backend/mysql.rs:115`.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-db/README.md [hunk 25]
```diff
@@ -652,6 +658,32 @@ let recent = User::objects()
...
+        DateTruncKind::Day,
+        DateProjectionOrder::Asc,
+    )
+    .await?;
+let local_hours = User::objects()
+    .datetimes(
+        User::field_created_at(),
+        DateTimeTruncKind::Hour,
+        DateProjectionOrder::Desc,
+        Some(chrono_tz::Asia::Tokyo),
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-db/README.md:719]: Implemented in 7062602. The documented consumer dependency block now includes `chrono-tz`, matching the named-zone example; see `crates/reinhardt-db/README.md:250`.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-query/src/backend/mysql.rs [hunk 26]
```diff
@@ -92,6 +92,84 @@ impl MySqlQueryBuilder {
...
+		let zone = match time_zone {
+			Some(TemporalTimeZone::Named(zone)) => zone.clone(),
+			Some(TemporalTimeZone::Utc) | None => "+00:00".to_string(),
+		};
+		writer.push_value(
+			crate::value::Value::String(Some(Box::new(zone))),
+			|_index| self.placeholder(0),
+		);
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-query/src/backend/mysql.rs:111]: Implemented in 7062602. MySQL CREATE VIEW now inlines the embedded SELECT values, producing a literal temporal zone and an empty Values collection for the DDL statement while normal SELECTs remain parameterized; see `crates/reinhardt-query/src/backend/mysql.rs:1630` and `crates/reinhardt-query/tests/temporal_projection.rs:267`.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-query/src/backend/postgres.rs [hunk 27]
```diff
@@ -88,11 +88,63 @@ impl PostgresQueryBuilder {
...
+		} else {
+			self.write_simple_expr(writer, expr);
+		}
+		if output == TemporalTruncOutput::DateTime {
+			writer.push(" AT TIME ZONE ");
+			let zone = match time_zone {
+				Some(TemporalTimeZone::Named(zone)) => zone.clone(),
+				Some(TemporalTimeZone::Utc) | None => "UTC".to_string(),
+			};
+			self.write_temporal_zone(writer, zone, unquoted);
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-query/src/backend/postgres.rs:126]: Implemented in 7062602. The public temporal-expression contract now explicitly requires a `TIMESTAMP WITH TIME ZONE` source for PostgreSQL datetime output and requires callers to convert naive timestamps explicitly; this avoids silently applying session-zone semantics where `SimpleExpr` has no physical-column metadata. See `crates/reinhardt-query/src/expr/func.rs:67` and `crates/reinhardt-query/src/expr/simple_expr.rs:170`.

🤖 Generated with [Codex](https://openai.com/codex)

File: crates/reinhardt-query/tests/temporal_projection.rs [hunk 28]
```diff
@@ -0,0 +1,223 @@
...
+			"\"occurred_on\") AS INTEGER) + 6) % 7) || ' days')"
+		)
+		.to_string(),
+		TemporalTruncKind::Day => "DATE(\"occurred_on\")".to_string(),
+		_ => panic!("date matrix uses only date truncation kinds"),
+	}
+}
+
+#[test]
+fn date_projection_sql_covers_every_kind_and_order_for_all_backends() {
```
[the remark this answers is not in the document]
Comment (kent8192) [reply crates/reinhardt-query/tests/temporal_projection.rs]: Implemented in 7062602. Both new projection suites import `rstest` and apply `#[rstest]` to every synchronous and async test; see `crates/reinhardt-query/tests/temporal_projection.rs:2` and `crates/reinhardt-db/tests/date_projection.rs:15`.

🤖 Generated with [Codex](https://openai.com/codex)
